include/BSMPT/minimum_tracer/minimum_tracer.h

```diff
@@ -0,0 +1,827 @@
+   * to status string
+   * @param out int output of IsThereEWSymmetryRestoration
+   * @return status string for output
+   */
+  std::string GetStatusEWSR(const int &out);
+
+  /**
+   * @brief IsThereEWSymmetryRestoration checks if there is EW symmetry
+   * restoration at high temperatures
+   * @return status; if status < -1: no minimum if status = 0: calculation
+   * failed, if status = 1: flat region with minimum, if status = 2: there is a
+   * minimum but no symmetry restoration, if status = 3: symmetry restoration
+   */
+  int IsThereEWSymmetryRestoration();
+
+  /**
+   * @brief SmallestEigenvalue calculate Eigenvalues of Hessian and returns
+   * smallest
+   * @param point point where to evaluate the Hessian
+   * @param Hessian Hessian function
+   * @return smallest Eigenvalue of Hessian
+   */
+  double SmallestEigenvalue(
+      const std::vector<double> &point,
+      const std::function<std::vector<std::vector<double>>(std::vector<double>)>
+          &Hessian);
+
+  /**
+   * @brief FindZeroSmallestEigenvalue
+   * @param point_1 first point
+   * @param T_1 temperature of first point
+   * @param point_2 second point
+   * @param T_2 temperature of second point
+   * @return stationary point found in between point_1 and point_2
+   */
+  std::vector<double> FindZeroSmallestEigenvalue(std::vector<double> point_1,
+                                                 double T_1,
+                                                 std::vector<double> point_2,
+                                                 double T_2);
+
+  /**
+   * @brief This function check if the gradient increses from point_1 to
+   * point_2. Not used at the moment.
+   *
+   * @param T is the temperature.
+   * @param point_1 initial point.
+   * @param point_2 final point.
+   * @param V is the potential.
+   * @param dV is the potential gradient.
+   * @return true if there is a barrier between the minima.
+   * @return false if there is no barrier between the minima.
+   */
+  bool GradientIncreases(
+      const double &T,
+      const std::vector<double> &point_1,
+      const std::vector<double> &point_2,
+      const std::function<double(std::vector<double>)> &V,
+      const std::function<std::vector<double>(std::vector<double>)> &dV);
+
+  /**
+   * @brief trackPhase with enforced global minimum tracing (= phase is checked
+   * if it is still the global minimum until it is no longer, then the current
+   * temperature is stored in globMinEndT)
+   * @param globMinEndT temperature at which phase is no longer global minimum
+   * @param point start point for tracking
+   * @param currentT start point temperature for phase tracking
+   * @param finalT end point temperature
+   * @param dT initial temperature step size
+   * @param output if true tracking output is printed on the screen
+   * @param unprotected if true we dont check the hessian
+   */
+  std::vector<Minimum> trackPhase(double &globMinEndT,
```

Comment:
```suggestion
  std::vector<Minimum> TrackPhase(double &globMinEndT,
```

The other functions start with a capital letter